packages/libs/react-ui/src/utils/useAnimation.ts

```diff
@@ -0,0 +1,113 @@
+import { useLayoutEffect } from '@react-aria/utils';
+import type { RefObject } from 'react';
+import { useCallback, useRef, useState } from 'react';
+import { flushSync } from 'react-dom';
+
+export function useEnterAnimation(
```

Comment:
can you add more documentation on the purpose of these utilities?

Reply:
those utils are ported from react-aria, they are used to check if the element has an animation and if it is ended/exiting or started/entering, I followed the implementation of react-aria-component, so if we need to add animation we can style using `[data-entring]` and `[data-exsiting]` for example currently the Popover support this pattern, but this is optional and we can get rid of it if not needed and since I wasn't sure if I need to implement those or not I implemented the behavior anyway, I will add js doc for those utils